### ISSUE TITLE
Making a model with a native question that was "transformed" with "Explore results" will lead to model with no metadata

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#45146]
### Describe the bug

In the process of hunting where metadata gets lots, I found a flow that might actually be a big culprit of why things don't work: if you make a native question look like a GUI question and then transform this question into a model, Metabase will treat this question as a GUI one, but without any metadata on it

### To Reproduce

1) new native question -> select * from people
2) click on "explore results", then transform it into a model
3) try to map the state field to the actual state field that exists on the table, you can't
![image](https://github.com/metabase/metabase/assets/1711649/bcb15b7d-2192-42bb-9a20-215f0e43b18f)
![image](https://github.com/metabase/metabase/assets/1711649/695564be-e5ea-4e9d-91d4-60726df5196b)

... now:
1) new native question -> select * from people
2) transform this into a model
3) see that the field now can be mapped to the state field in the table
![image](https://github.com/metabase/metabase/assets/1711649/13812e29-4654-49e7-a791-0ef6be5fe725)
![image](https://github.com/metabase/metabase/assets/1711649/325f6cd7-83cb-4a34-9ee9-27e65f3d135d)

### Expected behavior

We should know that the query is a native one and offer the metadata mapping

### Logs

NA

### Information about your Metabase installation

```JSON
I think it has always been like that
```

### Severity

P1

### Additional context

This might be a big reason why we lose Metadata and we don't know why, it might be that the users made a SQL question and masked it as a GUI one, and then we "believe" that metadata should be there as it seems like a normal GUI question

**Comment by paoliniluis** [comment]
moving into a P1, as this can cause real trouble

**Comment by metamben** [comment]
The cause of the issue is that 'Explore results" creates a new card based on the native card. If one turns the native card into a model, then mapping columns to database fields is possible. This is not possible if a card based on another (native) card is turned into a model. @luizarakaki what should we do, do you think? I've heard we want to move away from turning cards into models. If that's the case, then the use case might not be that important after all.

**Comment by perivamsi** [comment]
I agree with @luizarakaki's comments [here](https://metaboat.slack.com/archives/C04DN5VRQM6/p1721221208813019?thread_ts=1721155235.038309&cid=C04DN5VRQM6)

This seems to be by design. Not sure what the real issue is. 

@paoliniluis you can still add metadata, just not map columns - why do you think that is causing bugs as opposed to a limitation of product?

**Comment by paoliniluis** [comment]
@perivamsi and @luizarakaki 
The problem here is that users will:
1) create questions on top of this model
2) then they'll add those questions to dashboards and flters on those questions
3) they'll later find out that the filters don't have any filter values, and they'll have to re-do everything from scratch (since they won't be able to revert the question back to a SQL question/SQL model

it's a user flow which has a dead end and you end up with a huge pain
